src/mlpack/core/data/random_regression_generator.hpp

```diff
@@ -0,0 +1,210 @@
+  };
+  
+  ErrorParams(ErrorType type) : type(type) {};
+};
+
+/**
+ * @brief Class for generating linear regression data.
+ *
+ * This class provides a static function to generate synthetic regression data
+ * with specified parameters.
+ */
+class RegressionDataGenerator 
+{
+ public:
+  /**
+   * @brief Constructor for RegressionDataGenerator.
+   *
+   * @param nSamples Number of samples.
+   * @param nFeatures Number of features.
+   * @param ErrorParams Error distribution parameters.
+   * @param nTargets Number of target variables.
+   * @param intercept Bias term.
+   * @param sparsity Sparsity level of coefficients.
+   * @param outliersFraction Fraction of samples with outliers.
+   * @param outliersScale Scale of outliers.
+   * @param randomSeed Seed for random number generation.
+   */
+  RegressionDataGenerator(int nSamples,
+                          int nFeatures,
+                          const ErrorParams& errParams,
+                          int nTargets = 1,
+                          float intercept = 0.0,
+                          float sparsity = 0,
+                          float outliersFraction = 0,
+                          float outliersScale = 10,
+                          int randomSeed = 0): 
+        nSamples(nSamples),
+        nFeatures(nFeatures),
+        errParams(errParams),
+        nTargets(nTargets),
+        intercept(intercept),
+        sparsity(sparsity),
+        outliersFraction(outliersFraction),
+        outliersScale(outliersScale),
+        randomSeed(randomSeed)
+  {
+      ValidateParameters();
+  }
+
+  /**
+   * @brief Generate synthetic regression data.
+   *
+   * This function generates synthetic regression data with specified 
+   * parameters.
+   *
+   * @param X Matrix to store the generated features.
+   * @param y Matrix to store the generated response variable.
+   */
+  void GenerateData(arma::mat& X, arma::mat& y) const 
+  {
+    // Set the seed if needed
+    if (randomSeed)
+        arma::arma_rng::set_seed(randomSeed);
+    else
+        arma::arma_rng::set_seed_random();
+
+    // Generate random features with normal distribution
+    X.randn(nSamples, nFeatures);
+
+    // Generate coefficients with sparsity
+    arma::sp_mat coeff = arma::sprandn<arma::sp_mat>(nFeatures,
+                           nTargets, 1 - sparsity) * 100;
+    y = X * coeff + intercept;
+
+    // Add Noise
+    arma::mat error;
+
+    /// Generate noise based on the specified distribution
+    if (errParams.type == ErrorType::NormalDist) {
+        error = errParams.normalParams.mu + 
+                  errParams.normalParams.std * arma::randn(y.n_rows, y.n_cols);
+    }
+    else if (errParams.type == ErrorType::GammaDist) {
+      error = arma::randg(y.n_rows, y.n_cols, arma::distr_param(
+                errParams.gammaParams.alpha, errParams.gammaParams.beta));
+    }
+    else
+    {
+        throw std::invalid_argument("Invalid error distribution type. "
```

Comment:
1. Please fx the style so it is exactly the same as what mlpack is using.
2. Check other source code files for the styles, also we have a style guide somewhere in the github wiki.
3. Please also do the same on the test files,
4. Also add tests for float numbers, `arma::fmat`, and check other tests for this.